### DISCUSSION TITLE
Alertmanager multitenant UI with NGINX

### DISCUSSION THREAD

**Discussion by justpalma** [cortexproject/cortex#4807] [Q&A] [answered]
Hi, i'm currently tring to implement an NGINX configuration in order to show alertmanager UI for each tenant configured. I currently deployed cortex on k8s(EKS) and i have the followin tenant configured: dev1 and dev5.
I report below what is currently happening:
i have this nginx configuration
```
 location ~ /alertmanager {
           proxy_set_header X-Scope-OrgID dev1;
          proxy_pass      http://cma-cortex-alertmanager.cortex-aggregator.svc.cluster.local:8080$request_uri;
        }
```
and work fine. When i'm connecting to alertmanger, i can see only dev1 UI.
Of course, if i change header to dev5, i can see only dev5 UI.

Now i'm trying to enable both nginx configuration in order to have both UI accessible.

So i implemented this configuration
```
 location = /alert-dev1 {
          proxy_set_header X-Scope-OrgID dev1;
          proxy_pass_request_headers on;
          proxy_pass http://cma-cortex-alertmanager.cortex-aggregator.svc.cluster.local:8080/alertmanager;
        }

 location = /alert-dev5{
          
          proxy_set_header X-Scope-OrgID dev5;
          proxy_pass_request_headers on;
          proxy_pass http://cma-cortex-alertmanager.cortex-aggregator.svc.cluster.local:8080/alertmanager;
        }


 location ~ /alertmanager {
          proxy_pass      http://cma-cortex-alertmanager.cortex-aggregator.svc.cluster.local:8080$request_uri;
        }
```
But when the proxy_pass on /alertmanager is performed, the header is not set and alertmangaer UI return a black page with this message -> "no org id". Basically the header is not set.

Is it possible to configure multiple alertmanager UI and what is the best way to configure nginx?

**Comment by friedrichg** [comment, marked as answer]
Maybe something like this could help:

```
location ~ ^/(dev[0-9]+)/alertmanager/?((?<=/).*)?$ {
  proxy_set_header X-Scope-OrgID $1;
  proxy_pass  http://cma-cortex-alertmanager.cortex-aggregator.svc.cluster.local:8080/alertmanager/$2$is_args$args;
}
```
then the url for alertmanagers is 
https://cortex.example.com/dev1/alertmanager
https://cortex.example.com/dev5/alertmanager
etc...

**Reply by justpalma** [reply]
ok, now it's working.
I'm still trying to understand why that works.

If i understand right, in $1 i have the first path of the url(in our case it will be dev1/dev5/dev... ecc) but i'm not sure what is inside $2 $is_args and $args.

But at least it works 💯
